File: android/lib/models/occasion.dart
class Occasion {
    final int id;
    final String name;
    final String thumbnail;
    final String bannerImage;
    final DateTime startDate;
    final DateTime endDate;

    Occasion({
      required this.id,
      required this.name,
      required this.thumbnail,
      required this.bannerImage,
      required this.startDate,
      required this.endDate,
    });

    factory Occasion.fromJson(Map<String, dynamic> json) {
      return Occasion(
        id: json['id'],
        name: json['name'],
        thumbnail: json['thumbnail'],
        bannerImage: json['bannerImage'],
        startDate: DateTime.parse(json['startDate']),
        endDate: DateTime.parse(json['endDate']),
      );
    }
}

File: backend/dalathasfarm/src/main/java/com/example/dalathasfarm/models/Occasion.java
package com.example.dalathasfarm.models;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "occasions")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Occasion {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false, unique = true)
    private String name;

    @Column(unique = true)
    private String thumbnail;

    private String description;

    @Column(name = "start_date")
    private LocalDate startDate;

    @Column(name = "end_date")
    private LocalDate endDate;

    @Column(name = "banner_image")
    private String bannerImage;

    @Column(name = "is_active")
    private Boolean isActive;
}
